src/xSTIR/pSTIR/STIR.py

```diff
@@ -470,6 +474,16 @@ def set_modality(self, mod):
         """
         return parms.set_char_par(self.handle, 'ImageData', 'modality', mod)
 
+    @property
+    def __array_interface__(self):
+        """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
+        return {'shape': self.shape, 'typestr': '<f4', 'version': 3,
+                'data': (parms.size_t_par(self.handle, 'ImageData', 'address'), False)}
+
+    def asarray(self):
+        """Returns view of self"""
+        return numpy.asarray(self)
```

Comment:
More array_api_compat...

```suggestion
    def asarray(self, xp=numpy):
        """Returns view of self"""
        return xp.asarray(self)
```

Reply:
ok